metabase/metabase
# Migrate NewsletterForm to formik

Epic https://github.com/metabase/metabase/issues/26178

Migrates `NewsletterForm` from our form framework to `formik`:
- Several unneeded wrappers were removed because it's now possible to style form components directly
- As the form doesn't rely on `redux` anymore it was possible to remove jest mocks and add tests that it's submitted with the correct value
- Added `Email` prefix to styled components to differentiate between shared form components and ones specific to this form

How to test:
- Run a fresh Metabase instance
- Complete setup. The last step should prompt you to subscribe to the newsletter (this form).
- Make sure it is possible to enter an email and click Subscribe. There should be a request fired to a mail server.

<img width="727" alt="Screenshot 2022-11-01 at 12 59 40" src="https://user-images.githubusercontent.com/8542534/199306493-eafdd4c1-0e56-4de5-aabc-2889881461fc.png">
